src/biotite/interface/rdkit/mol.py

```diff
@@ -110,24 +119,29 @@ def to_mol(
     HB3
     HXT
     """
+    if not explicit_hydrogen:
+        atoms = atoms[..., atoms.element != "H"]
     mol = EditableMol(Mol())
 
     has_charge_annot = "charge" in atoms.get_annotation_categories()
     for i in range(atoms.array_length()):
         rdkit_atom = Atom(atoms.element[i].capitalize())
         if has_charge_annot:
             rdkit_atom.SetFormalCharge(atoms.charge[i].item())
+        if explicit_hydrogen:
+            rdkit_atom.SetNoImplicit(True)
```

Comment:
In the base case where `explicity_hydrogen=True`, what would this mean if I just call `to_mol` for a typical crystal structure that won't have any hydrogens resolved? Will RDKit infer charges / valences in that case? If so this might lead to broken molecules -- if that were the case, should we check that if explicit hydrogen is true there **have** to be hydrogens in the structure?

Reply:
Charges would also not be inferred automatically before this PR. But I am not sure what this means for valence: As the bond types are also explicitly set I guess *RDKit* assumes a radical, if `explicit_hydrogen=True`, but hydrogen atoms are actually missing. Probably, I should test this.

Having a check there sounds like a good idea, especially as I would agree that this could be a common mistake. However, I also think strictly checking for the simple presence of hydrogen atoms might not be sensible enough, as there are valid molecules without hydrogen atoms, although they appear rarely. What do you think about raising a warning as a 'reminder' to the user to check the input?